config/error-prone-suppressions/compile-phase-suppressions.xml

```diff
@@ -13,4 +13,10 @@
     <description>Avoid `Collectors.to{List,Map,Set}` in favor of collectors that emphasize (im)mutability</description>
     <lineContent>}).collect(Collectors.toSet()));</lineContent>
   </error>
+  <error>
+    <sourceFile>SarifLogger.java</sourceFile>
+    <bugPattern>IdentityConversion</bugPattern>
+    <description>Avoid or clarify identity conversions</description>
+    <lineContent>.replace(VERSION_PLACEHOLDER, String.valueOf(version))</lineContent>
```

Comment:
can you share reason of suppression in comment above?
We always do this to let next reader of this code understand our position or reasoning of ignoring best practices.

Reply:
It's not an ideal situation but the `version` here can be `null` while the replace will throw an NPE if it receives `null. Let me know if I should update the comment.

Reply:
in change of code:
```
✔ ~/java/github/romani/checkstyle [master|✚ 1⚑ 2] 
06:38 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java b/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java
index 82e848b..c8cdecb 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java
:...skipping...
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java b/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java
index 82e848b..c8cdecb 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/SarifLogger.java
@@ -31,6 +31,8 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
 
+import org.apache.commons.lang3.StringUtils;
+
 import com.puppycrawl.tools.checkstyle.api.AuditEvent;
 import com.puppycrawl.tools.checkstyle.api.AuditListener;
 import com.puppycrawl.tools.checkstyle.api.SeverityLevel;
@@ -140,7 +142,7 @@ public class SarifLogger extends AbstractAutomaticBean implements AuditListener
     public void auditFinished(AuditEvent event) {
         final String version = SarifLogger.class.getPackage().getImplementationVersion();
         final String rendered = report
-            .replace(VERSION_PLACEHOLDER, String.valueOf(version))
+            .replace(VERSION_PLACEHOLDER, StringUtils.defaultString(version))
             .replace(RESULTS_PLACEHOLDER, String.join(",\n", results));
         writer.print(rendered);
         if (closeStream) {

```
test failures:
```
diff (-expected +actual):
    @@ -13,8 +13,8 @@
               "organization": "Checkstyle",
               "rules": [
               ],
    -          "semanticVersion": "null",
    -          "version": "null"
    +          "semanticVersion": "",
    +          "version": ""
```

Reply:
I do not know if such fields ok to me empty if not properly defined.
https://github.com/checkstyle/checkstyle/blob/a2ba2b27bc7d59f9619d96663881f000260a913f/src/test/resources/com/puppycrawl/tools/checkstyle/sariflogger/ExpectedSarifLoggerLineOnly.sarif#L1-L20

I am ok to keep suppression.